x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/service/RestDeleteServiceAccountTokenAction.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.rest.action.service;
+
+import org.elasticsearch.action.support.WriteRequest;
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.license.XPackLicenseState;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.action.RestToXContentListener;
+import org.elasticsearch.xpack.core.security.action.service.DeleteServiceAccountTokenAction;
+import org.elasticsearch.xpack.core.security.action.service.DeleteServiceAccountTokenRequest;
+import org.elasticsearch.xpack.security.rest.action.SecurityBaseRestHandler;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.elasticsearch.rest.RestRequest.Method.DELETE;
+
+public class RestDeleteServiceAccountTokenAction extends SecurityBaseRestHandler {
+
+    public RestDeleteServiceAccountTokenAction(Settings settings, XPackLicenseState licenseState) {
+        super(settings, licenseState);
+    }
+
+    @Override
+    public List<Route> routes() {
+        return List.of(
+            new Route(DELETE, "/_security/service/{namespace}/{service}/credential/token/{name}"));
```

Comment:
This URL form is what we agreed upon in the design doc. But I wonder whether we would need to support bulk deleting the service tokens in future? If so, is it better to accept a payload like: `{"names": []}`. It is nice to have the bodyless URL form as well. So maybe best of both worlds would be like:
* If `{name}` is provided, disallow any payload and proceed to delete a single token
* If `{name}` is *not* provided, mandate a request body of `{"names": []}`

Reply:
I would hope that true bulk is not needed.
We're anticipating < 10 tokens per service, so calling a delete for each one seems reasonable.

I think a bulk version would be fine in theory, but I don't think it's worth adding since we don't have a clear need for it.